tests/runtime/main.rs

```diff
@@ -563,10 +564,10 @@ fn tests(name: &str, dir_name: &str) -> Result<Vec<PathBuf>> {
 
             cmd.current_dir(&out_dir);
 
-            cmd.arg("build")
+            cmd.arg("publish")
                 .arg(out_dir.join(format!("{camel}.csproj")))
                 .arg("-c")
-                .arg("Debug")
+                .arg(configuration)
                 .arg("/p:PlatformTarget=AnyCPU")
                 .arg("--self-contained")
```

Comment:
for me `--self-contained` doesn't' work with `9.0.100-preview.6.24328.19` without workload. The publish doesn't fail but the file is located elsewhere and it's not single file.

After I install workload `Microsoft.NET.Runtime.WebAssembly.Wasi.Sdk version 9.0.0-preview.6.24327.7` I get
```
FooWorld failed with 4 error(s) (11.4s) → xxx\
    wasm-ld : error : unknown argument: --component-type
    wasm-ld : error : unknown file type: FooWorld_component_type.wit
```

Which makes sense for preview6.

With preview7 I get
```
    1: error while executing at wasm backtrace:
           0: 0x296a7b - .tmpJCkP17!fseek
           1: 0x8efe - .tmpJCkP17!load_icu_data
           2: 0x8ff4 - .tmpJCkP17!mono_wasm_load_runtime
           3: 0x9299 - .tmpJCkP17!initialize_runtime
           4: 0x92b8 - .tmpJCkP17!main
           5: 0x28bc5c - .tmpJCkP17!__main_void
           6: 0x7e78 - .tmpJCkP17!_start
           7: 0xb40aff - wit-component:adapter:wasi_snapshot_preview1!wasi:cli/run@0.2.0#run
```

Reply:
nothihng about this will work without the workload

Reply:
I had to add `<WasmBuildNative>true</WasmBuildNative>`  to get it to produce a single file